src/main/scala/edu/ie3/powerFactory2psdm/config/validate/conversion/ConversionModeValidators.scala

```diff
@@ -0,0 +1,173 @@
+
+import edu.ie3.powerFactory2psdm.config.model.PvConversionConfig.{
+  PvFixedFeedIn,
+  PvModelConversionMode,
+  PvModelGeneration
+}
+import edu.ie3.powerFactory2psdm.config.model.WecConversionConfig.{
+  WecFixedFeedIn,
+  WecModelConversionMode,
+  WecModelGeneration
+}
+import edu.ie3.powerFactory2psdm.config.validate.ConfigValidator.{
+  validateCpCharacteristic,
+  validateParameterSamplingMethod,
+  validateQCharacteristic
+}
+import edu.ie3.powerFactory2psdm.exception.io.ConversionConfigException
+
+import scala.util.{Failure, Success}
+
+object ConversionModeValidators {
+
+  object PvConversionModeValidator
+      extends ConversionModeValidator[PvModelConversionMode] {
+
+    def apply(conversionMode: PvModelConversionMode): Unit = {
+      conversionMode match {
+        case x: PvFixedFeedIn     => validateQCharacteristic(x.qCharacteristic)
+        case x: PvModelGeneration => validateModelGenerationParams(x)
+      }
+    }
+
+    private[config] def validateModelGenerationParams(
+        params: PvModelGeneration
+    ): Unit = {
+      validateParameterSamplingMethod(params.albedo, 0, 1) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            s"The albedo of the plants surrounding: ${params.albedo} isn't valid.",
+            exc
+          )
+      }
+      validateParameterSamplingMethod(params.azimuth, -90, 90) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            s"The azimuth of the plant: ${params.azimuth} isn't valid.",
+            exc
+          )
+      }
+      validateParameterSamplingMethod(params.etaConv, 0, 100) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            s"The efficiency of the plants inverter: ${params.azimuth} isn't valid.",
+            exc
+          )
+      }
+      validateParameterSamplingMethod(params.kG, 0, 1) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            s"The PV generator correction factor (kG): ${params.kG} isn't valid.",
+            exc
+          )
+      }
+      validateParameterSamplingMethod(params.kT, 0, 1) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            s"The PV temperature correction factor (kT): ${params.kT} isn't valid.",
+            exc
+          )
+      }
+      validateQCharacteristic(params.qCharacteristic) match {
+        case Success(_) =>
+        case Failure(exc) =>
+          throw ConversionConfigException(
+            "The PV q characteristic configuration isn't valid.",
+            exc
+          )
+      }
+    }
+
+  }
+  object WecConversionModeValidator
+      extends ConversionModeValidator[WecModelConversionMode] {
+
+    def apply(conversionMode: WecModelConversionMode): Unit =
```

Comment:
you may call this method `validate(...)` instead of `apply(...)` as by convention apply is the factory method and not a regular method ....